aptos-move/framework/aptos-framework/sources/Coin.move

```diff
@@ -85,8 +86,56 @@ module AptosFramework::Coin {
         borrow_global<CoinInfo<CoinType>>(coin_address).supply
     }
 
+    /// Returns the `value` of the passed in `coin`. 
+    public fun value<CoinType>(coin: &Coin<CoinType>): u64 {
+        coin.value
+    }
+
+    /// Returns `true` if the type `CoinType` is a registered coin.
+    /// Returns `false` otherwise.
+    public fun is_coin<CoinType>(): bool {
+        let type_info = TypeInfo::type_of<CoinType>();
+        let coin_address = TypeInfo::account_address(&type_info);
+        exists<CoinInfo<CoinType>>(coin_address)
+    }
+
     // Public functions
 
+    /// Create a new `Coin<CoinType>` with a value of `0`.
+    public fun zero<CoinType>(): Coin<CoinType> {
+        Coin<CoinType> {
+            value: 0
+        }
+    }
+
+    /// Destroy a zero-value coin. Calls will fail if the `value` in the passed-in `token` is non-zero
+    /// so it is impossible to "burn" any non-zero amount of `Coin` without having
+    /// a `BurnCapability` for the specific `CoinType`.
+    public fun destroy_zero<CoinType>(zero_coin: Coin<CoinType>) {
+        let Coin { value } = zero_coin;
+        assert!(value == 0, Errors::invalid_argument(EDESTRUCTION_OF_NONZERO_TOKEN))
+    }
+
+    /// Extract `amount` from the passed-in `coin`, where the original token is modified in place.
+    public fun extract<CoinType>(coin: &mut Coin<CoinType>, amount: u64): Coin<CoinType> {
+        assert!(coin.value >= amount, Errors::invalid_argument(EINSUFFICIENT_BALANCE));
+        coin.value = coin.value - amount;
+        Coin { value: amount }
+    }
+
+    /// Removes `amount` of value from the passed in `coin` and returns original coin and new one contains `amount`.
+    public fun split<CoinType>(coin: Coin<CoinType>, amount: u64): (Coin<CoinType>, Coin<CoinType>) {
```

Comment:
do you have specific use case for this in mind? I thought most case `extract` is enough

Reply:
Hey, good question.

We used `split`, but tbh i think it's just matter of taste and code style preferences. 

I removed `split`, because at this stage i think you are right and it's too early, better if we get feedback from devs first on existing functions which require mutable references.